rest/route/test.go

```diff
@@ -72,55 +79,138 @@ func (tgh *testGetHandler) Parse(ctx context.Context, r *http.Request) error {
 }
 
 func (tgh *testGetHandler) Run(ctx context.Context) gimlet.Responder {
-	tests, err := tgh.sc.FindTestsByTaskId(tgh.taskId, tgh.key, tgh.testName, tgh.testStatus, tgh.limit+1, tgh.testExecution)
+	var (
+		tests []testresult.TestResult
+		key   string
+	)
+
+	// Check cedar test results first.
+	dbTask, err := task.FindByIdExecution(tgh.taskID, &tgh.testExecution)
 	if err != nil {
-		return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "Database error"))
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "database error"))
+	} else if dbTask == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			Message:    "not found",
+			StatusCode: http.StatusNotFound,
+		})
+	}
+	opts := apimodels.GetCedarTestResultsOptions{
+		BaseURL:   evergreen.GetEnvironment().Settings().Cedar.BaseURL,
+		Execution: tgh.testExecution,
+	}
+	if len(dbTask.ExecutionTasks) > 0 {
+		opts.DisplayTaskID = tgh.taskID
+	} else {
+		opts.TaskID = tgh.taskID
 	}
+	cedarTestResults, err := apimodels.GetCedarTestResults(ctx, opts)
+	if err != nil {
+		grip.Warning(message.WrapError(err, message.Fields{
+			"task_id": tgh.taskID,
+			"message": "problem getting cedar test results",
+		}))
 
+		// No cedar test results, fall back to the database.
+		tests, err = tgh.sc.FindTestsByTaskId(tgh.taskID, tgh.key, tgh.testName, tgh.testStatus, tgh.limit+1, tgh.testExecution)
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "database error"))
+		} else if len(tests) == 0 {
+			return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+				Message:    "not found",
+				StatusCode: http.StatusNotFound,
+			})
+		}
+
+		lastIndex := len(tests)
+		if lastIndex > tgh.limit {
+			key = string(tests[tgh.limit].ID)
+			lastIndex = tgh.limit
+		}
+		// Truncate the hosts to just those that will be returned.
+		tests = tests[:lastIndex]
+	} else {
+		startAt, err := strconv.Atoi(tgh.key)
+		if err != nil {
+			return gimlet.MakeJSONErrorResponder(errors.New("invalid start_at"))
+		}
+
+		var filteredCount int
+		cedarTestResults, filteredCount = graphql.FilterSortAndPaginateCedarTestResults(
+			cedarTestResults,
+			tgh.testName,
+			[]string{tgh.testStatus},
+			"",
+			1,
+			startAt,
+			tgh.limit,
+		)
+
+		if startAt*tgh.limit < filteredCount {
+			key = string(startAt + 1)
+		}
+	}
+
+	return tgh.buildResponse(cedarTestResults, tests, key)
+}
+
+func (tgh *testGetHandler) buildResponse(cedarTestResults []apimodels.CedarTestResult, testResults []testresult.TestResult, key string) gimlet.Responder {
```

Comment:
If `buildResponse` devolves into `addDataToResponse`, which accepts an `interface`, can't we just have it accept just one type of test result of type `interface{}` so `buildResponse()` can receive either `apimodels.CedarTestResult` or `testresult.TestResults` but not both? That way, we don't unnecessarily have to pass in both of types of test results when only one is set.

Reply:
It would actually be more code to do it that way because then I would have to pass in a slice of `interface{}` which would require creating a slice of `*apimodels.CedarTestResult` or `*testresult.TestResults` and, in doing so, looping through each test result separately in `Run`.

Reply:
And also slightly less efficient since we would have to loop through the test result slice twice.